test/api/encoding.js

```diff
@@ -12,14 +12,11 @@ function fixturePath(fixture) {
   return path.resolve(__dirname, "fixtures/encoding", fixture);
 }
 
-// This can't use `fs.promises`, as it has a bug with non-UTF-8 encodings:
-let readFile;
+// This can't use `fs.promises`, as it leads to non-UTF-8 files getting interpreted
+// as UTF-8 when passed to `ArrayBuffer`, `DataView` or `Int8Array` due to
+// buffer overflow issues.
```

Comment:
Is there a bug filed for this anywhere? If so, please link it here. (If not you should probably file a bug with Node.js.)

Reply:
Or is this only a bug with old versions of Node.js? If so, also put that info in the comment.

Reply:
Ping on adding a comment pointing to the Node.js bug report.

Reply:
I’ve opened <https://github.com/nodejs/node/issues/32302>.

Reply:
Unfortunately it's still not clear from that bug report whether the problem is a Node.js bug or a bug in your code/in jsdom. As I commented over there, we need a minimal reproduction case (not involving jsdom) before we can accept this as a Node bug.

Reply:
Well, <https://github.com/nodejs/node/issues/32302> has now been accepted as a **Node** bug.

Reply:
That does not appear to be what has happened.